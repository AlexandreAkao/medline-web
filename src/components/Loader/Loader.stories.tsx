import { ComponentStory, ComponentMeta } from '@storybook/react';

import Loader from 'components/Loader';
import colors from 'styles/colors';

export default {
  title: 'Loader',
  args: {},
  component: Loader,
  decorators: [
    Story => (
      <div
        style={{
          background: colors.primary.normal,
          width: '100%',
          height: '100%',
        }}
      >
        <Story />
      </div>
    ),
  ],
} as ComponentMeta<typeof Loader>;

const Template: ComponentStory<typeof Loader> = () => <Loader />;

export const Primary = Template.bind({});
Primary.args = {};
