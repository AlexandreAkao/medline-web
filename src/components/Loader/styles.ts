import styled, { keyframes } from 'styled-components';

import colors from 'styles/colors';
import zIndexPriority from 'styles/zIndexPriority';
import { getColorAlpha } from 'utils/colors';

const spinner = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
`;

export const LoaderContainer = styled.div`
  z-index: ${zIndexPriority.MAX};
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: ${getColorAlpha(colors.black.soft, 0.6)};
`;

export const LoaderSpin = styled.div`
  position: absolute;
  left: 50%;
  top: 50%;
  width: 100px;
  height: 100px;
  border: 15px solid ${colors.white.lighter};
  border-top: 15px solid ${colors.black.soft};
  border-radius: 50%;
  animation: ${spinner} 1.5s linear infinite;
`;
