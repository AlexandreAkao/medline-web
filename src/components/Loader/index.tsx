import { LoaderContainer, LoaderSpin } from 'components/Loader/styles';

function Loader() {
  return (
    <LoaderContainer>
      <LoaderSpin />
    </LoaderContainer>
  );
}

export default Loader;
