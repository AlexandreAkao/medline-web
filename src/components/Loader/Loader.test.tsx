import { render, screen } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import * as stories from 'components/Loader/Loader.stories';
import Loader from 'components/Loader';

const { Primary } = composeStories(stories);

describe('Loader', () => {
  const renderComponent = (customProps: Partial<ILoaderProps> = {}) => {
    const props: ILoaderProps = {
      isVisible: true,
      ...customProps,
    };
    return render(<Loader {...props} />);
  };

  it('should match with snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render a Loader', () => {
    render(<Primary />);
    const spinElement = screen.getByTestId(/test-loader-spin/i);
    expect(spinElement).toBeInTheDocument();
  });

  it('should not render Loader', () => {
    const { container } = renderComponent({ isVisible: false });
    expect(container.childElementCount).toBeFalsy();
  });
});
