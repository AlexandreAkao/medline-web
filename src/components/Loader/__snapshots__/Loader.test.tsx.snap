// Vitest Snapshot v1

exports[`Loader > should match with snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ gYwlUD"
  >
    <div
      class="sc-gsnTZi hCaNjM"
      data-testid="test-loader-spin"
    />
  </div>
</div>
`;
